File: Java/Address/Address.java
//********************************************************************
//  Address.java       Author: Lewis/Loftus
//
//  Represents a street address.
//********************************************************************

public class Address
{
   private String streetAddress, city, state;
   private String zipCode;

   //-----------------------------------------------------------------
   //  Sets up this Address object with the specified data.
   //-----------------------------------------------------------------
   public Address (String street, String town, String st, String zip)
   {
      streetAddress = street;
      city = town;
      state = st;
      zipCode = zip;
   }

   //-----------------------------------------------------------------
   //  Returns this Address object as a string.
   //-----------------------------------------------------------------
   public String toString()
   {
      String result;

      result = streetAddress + " ";
      result += city + ", " + state + " " + zipCode;

      return result;
   }
}

File: ObjectiveC/Address/Address/Address.m
//
//  Address.m
//  Address
//
//  Copyright © 2017 Warren Seto. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Address.h"

@implementation Address

/* @brief : Sets up this Address object with the specified data */
- (instancetype) initWithStreet:(NSString*)street town:(NSString*)town state:(NSString*)state andZip:(NSString*)zip
{
    self = [super init];
    
    _streetAddress = street;
    _city = town;
    _state = state;
    _zipCode = zip;
    
    return self;
}

/* @brief : Returns this Address object as a string */
- (NSString*) description
{
    NSString* result = [NSString stringWithFormat: @"%@ %@, %@ %@", _streetAddress, _city, _state, _zipCode ];
    return result;
}

@end
